### PR TITLE
Upgrade to Ionic 6

### LINKED ISSUE #1009: [BUG] Keyboard tab/voiceover navigation over modal does not work correctly
**Describe the bug**
After opening a modal and navigating using voice over through phone accessibility tools, the navigation goes over the elements in the underlying parent page, and not in the active modal.

**To Reproduce**

1. Activate voiceover/talkback
2. Open one of the modals on https://cookbook.kirby.design/examples/modal
3. Tab and listen to the underlying element being read.

**Kirby version**
- 1.2.11

**Expected behavior**
After opening a modal, the tab navigation should start inside the modal.

**Please complete the following information:**
- OS: Android
- Browser Chrome
- Browser version 81.0.4044.117

**Additional context**
Same as https://github.com/ionic-team/ionic/issues/17126

## Tasks:
- [ ] Move issue from "Ready to do" -> "in progress" on our [project board](https://github.com/kirbydesign/designsystem/projects/1)
- [ ] Create Fix Branch
- [ ] Create test reproducing the error
- [ ] Fix
- [ ] If relevant [Update Examples and Showcase](https://cookbook.kirby.design/home/showcase/button)
- [ ] Push commits to GitHub
- [ ] Open a pull request (remember to add `closes #issueno` to the description) and ask @kirbydesign/kirby-guild for review
- [ ] Merge to master and delete branch

### LINKED ISSUE #1932: [Housekeeping] Upgrade to Ionic 6
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the housekeeping request
Ionic v6 is out, and we should upgrade ASAP, before we introduce our own enhancements (to for example accessibility) on top of the newest improvements. 

https://ionicframework.com/docs/next/intro/upgrading-to-ionic-6.

## Have you considered any alternatives?
No! 😄

### LINKED ISSUE #2260: [BUG] Segmented Control is missing transition
...when navigating/selecting with keyboard (`SPACE`, `ENTER`) - press when control has focus.

Transition should be identical to when using mouse click.

## Screen recordings

First navigate using keyboard, then select using mouse:

https://user-images.githubusercontent.com/150451/168806280-b4de8044-3d69-491e-85ba-c0463c70041e.mov

https://user-images.githubusercontent.com/150451/168807262-c0214c1c-b464-4531-9a27-e5de8fea667a.mov

### PR DESCRIPTION
**Pull request by RasmusKjeldgaard**
## Which issue does this PR close?
Fixes #1932, fixes #1009, fixes #2260

## What is the new behavior?
Upgraded to the newest Ionic version at the time of speaking, and made the necessary changes as per the [upgrade guide](https://ionicframework.com/docs/intro/upgrading-to-ionic-6).

This is mostly style refactorings to the modal to account for the new shadow root. 
Furthermore, there are some changes to the Modal Animation Builder, as the modal was no longer animating correctly. 
The animation builder is essentially just holding a copy of the [enter/leave animations from Ionic](https://github.com/ionic-team/ionic-framework/tree/main/core/src/components/modal/animations), with some adjustments to account for removing the opacity on stacked backdrops. 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Are there any additional context?

With the new update we are seeing a boatload of these warnings in the console whenever a modal is opened: 

```
[Ionic Warning]: swipeToClose has been deprecated in favor of canDismiss.

If you want a card modal to be swipeable, set canDismiss to `true`. In the next major release of Ionic, swipeToClose will be removed, and all card modals will be swipeable by default.
```

I tried switching to canDismiss, but this does not seem to work for our drawer, as it is not a card modal. We will have to leave it to a followup issue to remove the deprecated functionality, before it is completely removed from Ionic. 

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Reminders
- [x] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [x] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [x] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [x] Request that the changes are code-reviewed 
- [ ] Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be merged to `develop` by Team Kirby.

### CODE REVIEW

File: libs/designsystem/src/lib/components/modal/services/action-sheet.helper.ts [hunk 1]
```diff
@@ -27,6 +27,10 @@ export class ActionSheetHelper {
       backdropDismiss: true,
     });
 
+    // Remove 'modal-shadow' element as we are no longer able to hide it with CSS in Ionic 6.
+    // We need to remove it so it does not obstruct the backdrop-click to dismiss functionality.
```
Comment (jkaltoft) [inline libs/designsystem/src/lib/components/modal/services/action-sheet.helper.ts:31]: 👍🏻 Helpful comment

**Review by jkaltoft — COMMENTED** [review]
In general the styling changes look sane. Nice with a bit of clean-up.

File: package.json [hunk 2]
```diff
@@ -93,7 +93,7 @@
     "@angular/platform-browser-dynamic": "13.3.9",
     "@angular/router": "13.3.9",
     "@fontsource/roboto": "4.2.1",
-    "@ionic/angular": "^5.5.2",
+    "@ionic/angular": "6.1.11",
```
Comment (MadsBuchmann) [inline package.json:96]: Is there a reason for not having the same version here as in `libs/designsystem/package.json`? (`^6.1.7`)

Comment (RasmusKjeldgaard) [reply package.json:96]: No, I forgot to update it there as well.

File: package.json [hunk 3]
```diff
@@ -93,7 +93,7 @@
     "@angular/platform-browser-dynamic": "13.3.9",
     "@angular/router": "13.3.9",
     "@fontsource/roboto": "4.2.1",
-    "@ionic/angular": "^5.5.2",
+    "@ionic/angular": "^6.1.7",
```
[the remark this answers is not in the document]
Comment (RasmusKjeldgaard) [reply package.json]: It seems like Ionic has been pretty solid when it comes to adhering to SemVer, but I have pinned it to a specific version now, so we don't need to worry.

File: libs/designsystem/src/lib/components/modal/services/action-sheet.helper.ts (hunk 1 above)
Comment (jkaltoft) [inline libs/designsystem/src/lib/components/modal/services/action-sheet.helper.ts:31]: Good comment 👍🏻

File: libs/designsystem/src/lib/components/modal/services/modal-animation-builder.service.ts [hunk 4]
```diff
@@ -65,7 +60,7 @@ export class ModalAnimationBuilderService {
           /**
            * Fallback for browsers that does not support `max()` (ex: Firefox)
            * No need to worry about statusbar padding since engines like Gecko
-           * are not used as the engine for standlone Cordova/Capacitor apps
+           * are not used as the engine for standalone Cordova/Capacitor apps
```
Comment (jkaltoft) [inline libs/designsystem/src/lib/components/modal/services/modal-animation-builder.service.ts:63]: giphub

**Review by jkaltoft — DISMISSED** [review]
Nice work, The Rasmus™

giphub

**Review by jkaltoft — DISMISSED** [review]

**Review by jkaltoft — APPROVED** [review]

**Pull request merged**